bitnami/etcd/templates/scripts-configmap.yaml

```diff
@@ -254,14 +254,27 @@ data:
     AUTH_OPTIONS="{{ $etcdAuthOptions }}"
     ETCDCTL_ENDPOINTS="{{range $i, $e := until $replicaCount }}{{ $etcdClientProtocol }}://{{ $etcdFullname }}-{{ $e }}.{{ $etcdHeadlessServiceName }}.{{ $releaseNamespace }}.svc.{{ $clusterDomain }}:{{ $clientPort }},{{ end }}"
     # Remove the last comma "," introduced in the string
-    export ETCDCTL_ENDPOINTS="$(sed 's/,/ /g' <<< $ETCDCTL_ENDPOINTS | awk '{$1=$1};1' | sed 's/ /,/g')"
+    export ETCDCTL_ENDPOINTS="${ETCDCTL_ENDPOINTS%%,}"
+    # prepare space separated list of endpoints
+    export etcdctl_endpoints_list="$(sed 's/,/ /g' <<< $ETCDCTL_ENDPOINTS)"
 
     mkdir -p "/snapshots" 1>&3 2>&4
-    if etcdctl $AUTH_OPTIONS endpoint health; then
-        echo "Snapshotting the keyspace..." 1>&3 2>&4
-        etcdctl $AUTH_OPTIONS snapshot save "/snapshots/db" 1>&3 2>&4
-    else
-        echo "Cluster isn't responding!!" 1>&3 2>&4
-        exit 1
-    fi
+
+    # walk over each node in the list, check if node healthy make snapshot and exit, otherwise try another node
+    for etcd_endpoint in $etcdctl_endpoints_list; do
```

Comment:
That's cool and will solve the problem, however, I'm wondering if we can avoid the loop. I mean, when we execute a snapshot whatever is the node, I presume, it is going to do a snapshot of the entire data. If this is the case and we do it three times then we are going to do two snapshots more, aren't we?

Reply:
I already thought about that, and that's why in line 270 there is `exit 0` statement, which ends the processing the whole script. This way we get only one snapshot.

There is no retry on failed snapshot execution but then the error processing defined in lines 237 will also end the script execution.

I've provided the example outputs in the description of the pull request to show how this works if for example first node is unavailable.